restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/client/api/ConferenceApiClient.java

```diff
@@ -0,0 +1,173 @@
+            onDownloaderResponse(message, self, sender);
+        } else if (message instanceof ReceiveTimeout) {
+            getContext().stop(self());
+        }
+    }
+
+    private void onDownloaderResponse(Object message, ActorRef self, ActorRef sender) {
+        final DownloaderResponse response = (DownloaderResponse) message;
+        if (logger.isInfoEnabled()) {
+            logger.info("Conference api response succeeded " + response.succeeded());
+            logger.info("Conference api response: " + response);
+        }
+        requestee.tell(message, self);
+        // since this is a one time disposable client lets clean it off after it sends DownloaderResponse.
+        getContext().stop(self());
+    }
+    protected void onStopConference(StopConference message, ActorRef self, ActorRef sender) throws URISyntaxException, ParseException {
+        requestee = sender;
+        conferenceDetailRecord = getConferenceDetailRecord();
+        if(conferenceDetailRecord == null){
+            logger.error("could not retrieve cdr by provided Sid");
+            onDownloaderResponse(new DownloaderResponse(new IllegalArgumentException("could not retrieve cdr by provided Sid")), self, sender);
+        } else {
+            try {
+                URI uri = new URI("/restcomm"+conferenceDetailRecord.getUri());
+                uri = UriUtils.resolve(uri);
+                if (logger.isInfoEnabled())
+                    logger.info("conference api uri is: "+uri);
+
+                Header[] headers = RestcommApiClientUtil.getBasicHeaders(message.getRequestingAccountSid(), storage);
+
+                ArrayList<NameValuePair> postParameters = new ArrayList<NameValuePair>();
+                postParameters.add(new BasicNameValuePair("Status", "Completed"));
+
+                httpAsycClientHelper = httpAsycClientHelper();
+                HttpRequestDescriptor httpRequestDescriptor =new HttpRequestDescriptor(uri, "POST", postParameters, -1, headers);
+                httpAsycClientHelper.tell(httpRequestDescriptor, self);
+            } catch (Exception e) {
+                logger.error("Exception while trying to terminate conference via api {} ", e);
+            }
+        }
+    }
+
+    private ConferenceDetailRecord getConferenceDetailRecord() throws ParseException{
+        if(conferenceSid == null){
+            ConferenceDetailRecordFilter filter = new ConferenceDetailRecordFilter(accountSid, "RUNNING%", null, null, name, 1, 0);
+            List<ConferenceDetailRecord> conferenceDetailRecords = storage.getConferenceDetailRecordsDao().getConferenceDetailRecords(filter);
+            if(conferenceDetailRecords == null || conferenceDetailRecords.isEmpty()){
+                logger.warning(String.format("could not retrive any conference record for this conference accountSid %s name: %s", accountSid, name));
+                return null;
+            }else{
+                return conferenceDetailRecords.get(0);
```

Comment:
@maria-farooq isn't this dangerous since we pick the first available Conference in the DB? I believe we should stop and fail the request when conferenceSid is null.

Reply:
`conferenceSid ` is not always available at local conference we care about `accountSid:name`

Reply:
@maria-farooq but still picking up the first on the list its dangerous since this one might not be the conference we are looking for

Reply:
@gvagenas as per our conference design/state, there is only one conference in DB in running state.

Reply:
@maria-farooq ??? Only one conference in running state? I am missing something here.

Reply:
There will be only 1 RUNNING Conference for AccountSid and given Name
Conference ID is unique because AccountSid+ConferenceName is unique